Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -2313,12 +2313,14 @@ internal void ChangePhaseOfInteraction(InputActionPhase newPhase, ref TriggerSta
             // Exception: if it was performed and we're to remain in started state, set the interaction
             //            to started. Note that for that phase transition, there are no callbacks being
             //            triggered (i.e. we don't call 'started' every time after 'performed').
-            if (newPhase == InputActionPhase.Performed && actionStates[actionIndex].interactionIndex != trigger.interactionIndex)
-            {
-                // We performed but we're not the interaction driving the action. We want to stay performed to make
-                // sure that if the interaction that is currently driving the action cancels, we get to perform
-                // the action. If we go back to waiting here, then the system can't tell that there's another interaction
-                // ready to perform (in fact, that has already performed).
+            if (newPhase == InputActionPhase.Performed &&
+                actionIndex != -1 && !actionStates[actionIndex].isPerformed &&
```

Comment:
Minor: Would prefer `actionIndex >= 0` for readability.

Reply:
changed to use the kInvalidIndex